test/e2e/specs/editor/plugins/register-block-type-hooks.spec.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * WordPress dependencies
+ */
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Register block type hooks', () => {
+	test.beforeEach( async ( { requestUtils, admin } ) => {
+		await requestUtils.activatePlugin(
+			'gutenberg-test-register-block-type-hooks'
+		);
+		await admin.createNewPost();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-register-block-type-hooks'
+		);
+	} );
+
+	test( 'has a custom category for Paragraph block', async ( { page } ) => {
+		await page.click( 'role=button[name="Toggle block inserter"i]' );
+
+		expect(
+			page.locator(
+				'.block-editor-block-types-list[aria-label="Widgets"] > .editor-block-list-item-paragraph'
```

Comment:
Thank you for working on another migration 😄 

It should be ok to use a role selector for this, I think it would be something like this, but I haven't tested to make sure it works:
```suggestion
				'role=listbox[name="Widgets"i] >> role=option[name="Paragraph"i]'
```